tests/config/format-test.js

```diff
@@ -40,6 +40,8 @@ const unstableTests = new Map(
     "typescript/prettier-ignore/mapped-types.ts",
     "js/comments/html-like/comment.js",
     "js/for/continue-and-break-comment-without-blocks.js",
+    // We updated to typescript 4.7 and this test fails (but this feature is not important).
+    "typescript/error-recovery/jsdoc_only_types.ts",
```

Comment:
Check `node.prefix` here https://github.com/prettier/prettier/blob/1eecc859852da911734b94c1092b6c41338fba52/src/language-js/print/typescript.js#L520 should be able to fix the ast.

Reply:
Thank you. This should be fixed in #12921. Please see [8ec843b](https://github.com/prettier/prettier/pull/12921/commits/8ec843b4e1507d5dc93d050263b0bf895dd63e32)